### DISCUSSION TITLE
Welcome to MinimalChessEngine Discussions!

### DISCUSSION THREAD

**Discussion by lithander** [lithander/MinimalChessEngine#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share critique and ideas for improvement.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lepned** [comment]
Hi,

Thanks for a very nice and minimalistic engine written in C#. I am trying to study the source code at the moment and find it very interesting indeed. Especially curious how you are able to search so deep from the start position. My engine (just started) only manages to search about 10-11 ply with AB search and transposition table enabled. Anyway, I found a problem with my engine in a simple pawn endgame which every school boy should know how to win but my engine was not able to solve it. In that spirit I tried the same position with minimalChessEngine with the same result, hmm.

Here is the position:

8/8/3k4/8/3K4/8/3P4/8 w - - 0 1

I guess it's connected to null move pruning for both of us since it involves zugzwang.

It would be interesting to get your view about what's going on here. Thanks in advance!
